### PR TITLE
[libmodbus] initial integration

### PR DESCRIPTION
**Pull request by 0x34d**
```libmodbus initial integration.```

#### Q: Way not libfuzzer
```
A: In One Word CLOSE_WAIT
```

### CODE REVIEW

**Comment by 0x34d** [comment]
@stephane Any Comment.

**Comment by jonathanmetzman** [comment]
> `libmodbus initial integration.`
> 
> #### Q: Way not libfuzzer
> ```
> A: In One Word CLOSE_WAIT
> ```

I'm sorry, I don't know what this means. Could you elaborate?

File: projects/libmodbus/fuzz/Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#Compiler
+LIBTOOL=libtool --mode=link --tag=CXX
+
+#Compiler Flags
+SET=cd ../ &&
+INC=-I../src/
+LIB=../src/libmodbus.la
+EXTCFLAGS= -Wall -Wextra
+CFLAGS=-g -O0 -fsanitize=address,undefined -fno-omit-frame-pointer
+LDFLAGS=-fsanitize=address,undefined
```
Comment (jonathanmetzman) [inline projects/libmodbus/fuzz/Makefile:16]: You shouldn't hardcode the sanitizers. You should use our CFLAGS which define them. This way our infra knows if it's running ASAN or UBSAN

**Review by jonathanmetzman — COMMENTED** [review]
We would accept this project if upstream agrees.

File: projects/libmodbus/fuzz/Makefile (hunk 1 above)
Comment (0x34d) [reply projects/libmodbus/fuzz/Makefile:16]: OK.

Comment (0x34d) [reply projects/libmodbus/fuzz/Makefile]: fixed.

**Comment by stephane** [comment]
I am pleased to have libmodbus tested with oss-fuzz. It is a subject that has been close to my heart for a long time and I have tested different tools in the past, but only in an ad hoc manner.

| Some time Client TCP Socket hung up in the CLOSE_WAIT stage.

I understand this sentence but what action do you expect from me on this issue?

**Comment by 0x34d** [comment]
> I am pleased to have libmodbus tested with oss-fuzz. It is a subject that has been close to my heart for a long time and I have tested different tools in the past, but only in an ad hoc manner.
> 
> > Some time Client TCP Socket hung up in the CLOSE_WAIT stage.
> 
> I understand this sentence but what action do you expect from me on this issue?

Mr. @stephane it's a kernel issue not in the project.
[READ](https://blog.cloudflare.com/this-is-strictly-a-violation-of-the-tcp-specification/)
Sometimes in fedora, this problem occurs.

**Comment by 0x34d** [comment]
@jonathanmetzman 
I Belive that all these changes fix everything.
If Anything New or a More radical approach is needed tell me.

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Pull request merged**